src/cache/tlru-cache.ts

```diff
@@ -32,32 +38,65 @@ export class TLRUCache extends LRUCache {
   }
 
   private evictExpiredCacheEntries() {
-    const now = new Date(Date.now());
     let cacheKey: string;
-    let evictionDate: Date;
 
     // Not using this.cache.forEach so we can break the loop once
     // we find the fist non-expired entry. Each entry after that
-    // is guaranteed to also be non-expired, because they are oldest->newest
-    for ([cacheKey, evictionDate] of this.cacheEntriesTTLRegistry.entries()) {
-      if (now >= evictionDate) {
+    // is guaranteed to also be non-expired, because iteration happens
+    // in insertion order
+    for (cacheKey of this.cache.keys()) {
+      if (!this.isCacheEntryValid(cacheKey)) {
         this.delete(cacheKey);
       } else {
         break;
       }
     }
   }
 
+  entries(): IterableIterator<[string, string]> {
+    this.evictExpiredCacheEntries();
+    return super.entries();
+  }
+
+  keys(): IterableIterator<string> {
+    this.evictExpiredCacheEntries();
+    return super.keys();
+  }
+
+  values(): IterableIterator<string> {
+    this.evictExpiredCacheEntries();
+    return super.values();
+  }
+
   delete(key: string): boolean {
     this.clearCacheEntryEvictionTimeIfExists(key);
     return super.delete(key);
   }
 
+  // has(key: string): boolean {
+  //   const hasValue = this.cache.has(key);
+  //
+  //   if (!this.isCacheEntryValid(key)) {
+  //     this.delete(key);
+  //     return false;
+  //   }
+  //
+  //   return hasValue;
+  // }
+
   get(key: string): string | undefined {
-    this.evictExpiredCacheEntries();
+    if (!this.cache.has(key)) {
+      return undefined;
+    }
+
+    const value = this.cache.get(key);
 
-    const value = super.get(key);
     if (value !== undefined) {
+      if (!this.isCacheEntryValid(key)) {
+        this.delete(key);
+        return undefined;
+      }
+
```

Comment:
otherwise, if valid, do we need to `this.cache.set(key, value);` to bump it to the bottom for the LRU like we do the parent class?

I wonder if we just check `!this.isCacheEntryValid(key)` first, if so delete it from the cache, then proceed to just call the parent method's `get()`

Reply:
> I wonder if we just check !this.isCacheEntryValid(key) first, if so delete it from the cache, then proceed to just call the parent method's get()

I've rewrote it to do just that + reset cache TTL timer if we got a cache hit from parent class